src/backend/catalog/index.c

```diff
@@ -950,8 +950,16 @@ index_create(Relation heapRelation,
 		}
 		else
 		{
-			indexRelationId =
-				GetNewRelFileNode(tableSpaceId, pg_class, relpersistence);
+			if (is_enr)
+			{
+				/* Index OIDs must be kept in normal OID range. */
+				indexRelationId = GetNewPermanentRelFileNode(tableSpaceId, pg_class, relpersistence);
```

Comment:
rather than adding a duplicate function, would it be easier to add a parameter to GetNewRelFileNode to indicate the same?

Reply:
I favored this method over adding a new parameter to avoid refactoring many parts of engine code with the new function